projects/uk.ac.man.cs.llvm/src/uk/ac/man/cs/llvm/ir/module/TargetDataLayout.java

```diff
@@ -33,7 +33,7 @@
 
     private final String layout;
 
-    private TargetDataLayout(String layout) {
```

Comment:
Can you use the static factory method `fromString` instead of making the constructor public?